server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverResponse.java

```diff
@@ -65,13 +65,27 @@
         declareAcknowledgedAndShardsAcknowledgedFields(PARSER);
     }
 
-    private String oldIndex;
-    private String newIndex;
-    private Map<String, Boolean> conditionStatus;
-    private boolean dryRun;
-    private boolean rolledOver;
-
-    RolloverResponse() {
+    private final String oldIndex;
+    private final String newIndex;
+    private final Map<String, Boolean> conditionStatus;
+    private final boolean dryRun;
+    private final boolean rolledOver;
+    // Needs to be duplicated, because shardsAcknowledged gets (de)serailized as last field whereas
+    // in other subclasses of ShardsAcknowledgedResponse this field (de)serailized as first field.
+    private final boolean shardsAcknowledged;
```

Comment:
This is the only anomaly when it comes to changing response classes over to use `Writable`, `shardsAcknowledged` field is serialized last here, and for all the other subclasses of `ShardsAcknowledgedResponse` this is serialized first. I found that the easiest solutions would be to re-introduce this variable here and overwrite the `isShardsAcknowledged ` method.